frontend/app/components/button.tsx

```diff
@@ -20,28 +25,57 @@ const variants = {
   red: 'border-red-700 bg-red-700 text-white hover:bg-red-800 focus:bg-red-800',
 } as const;
 
-type ButtonProps = ComponentProps<'button'> & {
+const baseClassName = 'inline-flex items-center justify-center rounded border align-middle font-lato outline-offset-4';
+
+export interface ButtonProps extends ComponentProps<'button'> {
+  endIcon?: ButtonEndIconProps['icon'];
+  endIconProps?: OmitStrict<ButtonEndIconProps, 'icon'>;
   pill?: boolean;
   size?: keyof typeof sizes;
+  startIcon?: ButtonStartIconProps['icon'];
+  startIconProps?: OmitStrict<ButtonStartIconProps, 'icon'>;
   variant?: keyof typeof variants;
-};
+}
 
 /**
  * Tailwind CSS Buttons from Flowbite
  * @see https://flowbite.com/docs/components/buttons/
  */
-export function Button({ children, className, disabled, pill, size = 'base', variant = 'default', ...props }: ButtonProps) {
-  const baseClassName = cn(
-    'inline-flex items-center justify-center rounded border align-middle font-lato no-underline outline-offset-4',
-    sizes[size],
-    variants[variant],
-    disabled && 'disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70',
-    pill && 'rounded-full',
-  );
-
-  return (
-    <button className={cn(baseClassName, className)} {...props}>
-      {children}
-    </button>
-  );
-}
+const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+  (
+    {
+      children,
+      className,
+      endIcon,
+      endIconProps,
+      pill,
+      size = 'base',
+      startIcon,
+      startIconProps,
+      variant = 'default',
+      ...props
+    },
+    ref,
+  ) => {
+    const disabledClassName = 'disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-70';
+    const buttonClassName = cn(
+      baseClassName,
+      sizes[size],
+      variants[variant],
+      disabledClassName,
+      pill && 'rounded-full',
+      className,
+    );
+    return (
+      <button className={buttonClassName} {...props} ref={ref}>
+        {startIcon && <ButtonStartIcon {...(startIconProps ?? {})} icon={startIcon} />}
+        {children}
+        {endIcon && <ButtonEndIcon {...(endIconProps ?? {})} icon={endIcon} />}
+      </button>
+    );
+  },
+);
+
+Button.displayName = 'Button';
+
+export { Button };
```

Comment:
Use `export function Button()` to be consistent with the rest of the application.